### DISCUSSION TITLE
What's the logic for packageRules conditions?

### DISCUSSION THREAD

**Discussion by Leksat** [renovatebot/renovate#8116] [Request Help] [answered]
I'm trying to prevent major updates for a certain package in a certain app of a monorepo:
```json5
  packageRules: [
    {
      updateTypes: ["major", "minor", "patch", "digest"],
      automerge: true,
    },
    {
      paths: ["apps/silverback-drupal/composer.json"],
      packageNames: [
        "drupal/core",
      ],
      major: {
        enabled: false,
      },
    }
  ],
```

According to the docs
> you can mix together both `packageNames` and `packagePatterns` in the same package rule and the rule will be applied if _either_ match

Not sure if this is applied to my case as well. I would like the rule to work only if both `paths` and `packageNames` match. Is this possible?

**Comment by rarkins** [comment, marked as answer]
> I would like the rule to work only if both `paths` and `packageNames` match. Is this possible?

Yes. The docs you quote refer to `packageNames` and `packagePatterns` and not about `paths`.

But don't nest `major` within the second rule. Use `updateTypes` instead, like you did in the first rule.

**Comment by Leksat** [comment]
Thank you for the superfast reply! Will do.

Also some feedback: For me it was (and still is) quite hard to understand Renovate config options. The docs are full of examples, but sometimes it isn't clear which option is a _condition_ and which is a _setting_.

Wouldn't it make sense to split them? For example:
```json5
  packageRules: [
    {
      conditions: {
        paths: [
          "apps/silverback-drupal/composer.json",
        ],
        packageNames: [
          "drupal/core",
        ],
      },
      settings: {
        updateTypes: [
          "minor",
          "patch",
          "digest",
        ],
      },
    },
  ],
```

(I'm not familiar with all Renovate options/features. Just an idea 🙂)

**Reply by viceice** [reply]
see #6805 and #6939 😏

**Reply by viceice** [reply]
btw: `updateTypes` is a condition too

**Comment by Leksat** [comment]
Just to be sure, this would be the final config to do what i want?
(Exclude major updates of `drupal/core` package in a certain app)
```
  packageRules: [
    {
      updateTypes: ["major", "minor", "patch", "digest"],
      automerge: true,
    },
    {
      paths: ["apps/silverback-drupal/composer.json"],
      packageNames: [
        "drupal/core",
      ],
      updateTypes: ["major"],
      enabled: false,
    }
  ],

**Reply by viceice** [reply]
Yes, but better use dependency dashboard with approval. So you can see the update there.

If the same package is used in other projects in same repo, you should use additional branch prefix to not block them.
